native-image-tests/local-scala/src/main/scala/com/lightbend/JacksonCheck.scala

```diff
@@ -0,0 +1,133 @@
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer
+import com.fasterxml.jackson.databind.ser.std.StdSerializer
+
+object JacksonCheck {
+
+  // from the akka-serialization-jackson docs
+
+  // type hierarchy / ADT
+  // FIXME: Still haven't figured out why @JsonCreator cannot be used here
+  final case class Zoo(@JsonProperty("primaryAttraction") primaryAttraction: Animal) extends JsonSerializable
+
+  // Note: native image needs additional marker trait on ADT supertype here or else
+  //       AkkaJacksonSerializationFeature has no way to detect and register the subtypes for
+  //       reflective access
+
+  @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+  @JsonSubTypes(
+    Array(
+      new JsonSubTypes.Type(value = classOf[Lion], name = "lion"),
+      new JsonSubTypes.Type(value = classOf[Elephant], name = "elephant"),
+      new JsonSubTypes.Type(value = classOf[Unicorn], name = "unicorn")))
+  sealed trait Animal extends JsonSerializable
+
+  final case class Lion(@JsonProperty("name") name: String) extends Animal
+  final case class Elephant(name: String, age: Int) extends Animal
+
+  @JsonDeserialize(`using` = classOf[UnicornDeserializer])
+  sealed trait Unicorn extends Animal
+  @JsonTypeName("unicorn")
+  case object Unicorn extends Unicorn
+
+  class UnicornDeserializer extends StdDeserializer[Unicorn](Unicorn.getClass) {
+    // whenever we need to deserialize an instance of Unicorn trait, we return the object Unicorn
+    override def deserialize(p: JsonParser, ctxt: DeserializationContext): Unicorn = Unicorn
+  }
+
+  // object enum
+  @JsonSerialize(`using` = classOf[DirectionJsonSerializer])
+  @JsonDeserialize(`using` = classOf[DirectionJsonDeserializer])
+  sealed trait Direction extends JsonSerializable
+
+  object Direction {
+    case object North extends Direction
+    case object East extends Direction
+    case object South extends Direction
+    case object West extends Direction
+  }
+
+  class DirectionJsonSerializer extends StdSerializer[Direction](classOf[Direction]) {
+    import Direction._
+
+    override def serialize(value: Direction, gen: JsonGenerator, provider: SerializerProvider): Unit = {
+      val strValue = value match {
+        case North => "N"
+        case East  => "E"
+        case South => "S"
+        case West  => "W"
+      }
+      gen.writeString(strValue)
+    }
+  }
+
+  class DirectionJsonDeserializer extends StdDeserializer[Direction](classOf[Direction]) {
+    import Direction._
+
+    override def deserialize(p: JsonParser, ctxt: DeserializationContext): Direction = {
+      p.getText match {
+        case "N" => North
+        case "E" => East
+        case "S" => South
+        case "W" => West
+      }
+    }
+  }
+
+  final case class Compass(@JsonProperty("currentDirection") currentDirection: Direction) extends JsonSerializable
+
+  // scala enums
+  // FIXME this one needs some very specific stuff for Scala enums, might not be worth it
```

Comment:
This fails with reflective access to `$outer` so to support it we'd need to detect `Enumeration` subtype in `AkkaJacksonSerializationFeature` and handle specifically

Reply:
Decided to give up on Scala stdlib enums for now and documented that those doesn't work